docx_editor/track_changes.py

```
@@ -214,6 +224,87 @@ def _apply_single_edit(self, op: EditOperation) -> int:
+            error = self._validate_single(op)
+            results.append(
+                EditValidationResult(
+                    index=i,
+                    paragraph=op.paragraph,
+                    valid=error is None,
+                    error=error,
+                )
+            )
+        return results
+
+    def _validate_single(self, op: EditOperation) -> str | None:
```

Ruff (0.15.20)

[warning] Too many return statements (10 > 6)

(PLR0911)